examples/cord_epsim/main.c

```diff
@@ -67,33 +69,45 @@ static gcoap_listener_t listener = {
 
 int main(void)
 {
+    char addrstr[IPV6_ADDR_MAX_STR_LEN];
     puts("Simplified CoRE RD registration example\n");
 
     /* fill riot info */
     sprintf(riot_info, "{\"ep\":\"%s\",\"lt\":%u}",
             cord_common_get_ep(), CORD_LT);
 
     /* parse RD address information */
-    sock_udp_ep_t rd_ep = {
-        .family    = AF_INET6,
-        .netif     = SOCK_ADDR_ANY_NETIF,
-        .port      = RD_PORT,
-    };
-
-    /* parse RD server address */
-    if (ipv6_addr_from_str((ipv6_addr_t *)&rd_ep.addr.ipv6, RD_ADDR) == NULL) {
+    sock_udp_ep_t rd_ep;
+    rd_ep.port = 0;
+
+    if (sock_udp_str2ep(&rd_ep, RD_ADDR) < 0) {
         puts("error: unable to parse RD address from RD_ADDR variable");
         return 1;
     }
 
+    /* if netif not specified in addr */
+    if ((rd_ep.netif == SOCK_ADDR_ANY_NETIF) && (gnrc_netif_numof() == 1)) {
```

Comment:
we should also handle the case for `gnrc_netif_numof() != 1` explicitly.

Reply:
but the check is only if we're using link-local address. gcoap has a similar check here:

https://github.com/RIOT-OS/RIOT/blob/4f6985612a44c4f96d660d7f5d54a287e5361851/examples/gcoap/gcoap_cli.c#L263-L266